doc/headers/linop.hpp

```diff
@@ -0,0 +1,126 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright 2017-2019
+
+Karlsruhe Institute of Technology
+Universitat Jaume I
+University of Tennessee
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its contributors
+   may be used to endorse or promote products derived from this software
+   without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+/**
+ * @defgroup LinOp Linear Operators
+ *
+ * @brief A module dedicated to the implementation and usage of the Linear
+ * operators in Ginkgo.
+ *
+ * Below we elaborate on one of the most important concepts of Ginkgo, the
+ * linear operator. The linear operator (LinOp) is a base class for all linear
+ * algebra objects in Ginkgo. The main benefit of having a single base class for
+ * the entire collection of linear algebra objects (as opposed to having
+ * separate hierarchies for matrices, solvers and preconditioners) is the
+ * generality it provides.
+ *
+ * @section linop_1 Linear operator as a concept.
```

Comment:
All of this is basically duplicating the LinOp documentation. Maybe: can you move a huge chunk of this into the LinOp class documentation itself and use doxygen `@copydoc` ?

Reply:
I think you can also stick the `@section` in the LinOp documentation, and maybe event the section 3, it probably would not hurt.